lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -0,0 +1,269 @@
+      ? new Date(dynamoRecord.timestamp) : undefined,
+    created_at: new Date(dynamoRecord.createdAt),
+    updated_at: new Date(dynamoRecord.updatedAt),
+  };
+
+  await knex(tableNames.granules).insert(updatedRecord);
+};
+
+/**
+ * Migrate File record from a Granules record from DynamoDB  to RDS.
+ *
+ * @param {ApiFile} file - Granule file
+ * @param {string} granuleId - ID of granule
+ * @param {string} collectionId - ID of collection
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migrateFileRecord = async (
+  file: ApiFile,
+  granuleId: string,
+  collectionId: string,
+  knex: Knex
+): Promise<void> => {
+  const [name, version] = collectionId.split('___');
+
+  const collectionCumulusId = await getRecordCumulusId<PostgresCollectionRecord>(
+    { name, version },
+    tableNames.collections,
+    knex
+  );
+
+  const granuleCumulusId = await getRecordCumulusId<PostgresGranuleRecord>(
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId },
+    tableNames.granules,
+    knex
+  );
+
+  const bucket = getBucket(file);
+  const key = getKey(file);
+
+  // Map old record to new schema.
+  const updatedRecord: PostgresFile = {
+    bucket,
+    key,
+    granule_cumulus_id: granuleCumulusId,
+    file_size: file.size,
+    checksum_value: file.checksum,
+    checksum_type: file.checksumType,
+    file_name: file.fileName,
+    source: file.source,
+    path: file.path,
+  };
+  await knex(tableNames.files).insert(updatedRecord);
+};
+
+/**
+ * Migrate granule and files from DynamoDB to RDS
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ * @param {Knex} knex
+ * @returns {Promise<MigrationSummary>} - Migration summary for files
+ */
+export const migrateGranuleAndFilesViaTransaction = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<MigrationSummary> => {
+  const fileMigrationSummary = {
+    dynamoRecords: 0,
+    success: 0,
+    failed: 0,
+    skipped: 0,
+  };
+
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(dynamoRecord, schemas.granule);
+  const files = dynamoRecord.files;
+  const granuleId = dynamoRecord.granuleId;
+  const collectionId = dynamoRecord.collectionId;
+
+  await knex.transaction(async () => {
+    await migrateGranuleRecord(dynamoRecord, knex);
+    await Promise.all(files.map(async (file : ApiFile) => {
+      fileMigrationSummary.dynamoRecords += 1;
+      try {
+        await migrateFileRecord(file, granuleId, collectionId, knex);
+        fileMigrationSummary.success += 1;
+      } catch (error) {
+        fileMigrationSummary.failed += 1;
+        logger.error(
+          `Could not create file record in RDS for file ${file}`,
+          error
+        );
+      }
+    }));
+  });
+
+  return fileMigrationSummary;
```

Comment:
I don't particularly like that this function returns `fileMigrationSummary` but couldn't think of a better way to propagate this information for `migrateGranulesAndFiles`